src/Orleans/Serialization/RemoteNonDeserializableException.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Runtime.Serialization;
+using Orleans.Runtime;
+
+namespace Orleans.Serialization
+{
+    using System;
+    using System.Text;
+
+    /// <summary>
+    /// Represents an exception which cannot be fully deserialized.
+    /// </summary>
+    [Serializable]
+    public class RemoteNonDeserializableException : OrleansException
+    {
+        public RemoteNonDeserializableException() { }
+
+        /// <summary>
+        /// Gets the type name of the original <see cref="Exception"/> represented by this instance.
+        /// </summary>
+        public string OriginalTypeName { get; internal set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the original type of this exception can be reconstructed during
+        /// deserialization.
+        /// </summary>
+        internal bool CanAttemptReconstruction { get; set; }
+
+        /// <summary>
+        /// Gets or sets the additional data deserialized alongside this 
+        /// </summary>
+        internal byte[] AdditionalData { get; set; }
```

Comment:
@jason-bragg I understand the skepticism - the value is questionable, but it does add value. This data is serialized regardless, but this property allows the exception to be safely round-tripped through nodes which cannot deserialize it fully, which is a possible (rare) case.

It allows the serializer to maintain the property that an exception of type X and an exception of type `RemoteNonDeserializableException` representing that original exception have an identical wire representation.